### DISCUSSION TITLE
Database migration from 5.9.7 to 5.10.9

### DISCUSSION THREAD

**Discussion by nathantg** [yamcs/yamcs#992] [Q&A]
I  am looking to migrate my yamcs server from an older 5.9.7 to the latest 5.10.9. It says in the release notes that 5.10.9 is backwards compatible with older versions of the database. It says that a rebuild of the parameter archive is required using (yamcs parameter-archive rebuild), but I am using the maven yamcs plugin to run yamcs. Is there a way to migrate the database when using the maven plugin?

**Comment by xpromache** [comment]
Shouldn't matter how you run Yamcs. You can install yamcs-cli with:
pip install --upgrade yamcs-cli

then
yamcs login
(select the instance)
yamcs parameter-archive purge
yamcs parameter-archive rebuild 2000-01-01T00:00:00 2100-01-01T00:00:00 

(sorry, we don't have a way to not provide a time range)

**Reply by nathantg** [reply]
I have an archived yamcs-data directory (from 5.9.6) that I am using when running yamcs (5.10.9). Is using that archived database for with yamcs as simple as copying the yamcs-data directory into YAMCS_HOME/target/yamcs?

Thanks for the response!

**Reply by xpromache** [reply]
yes, but I think it is better to set 
`dataDir: /another/directory
`
in yamcs.yaml because the target directory is removed when you run a mvn clean.

**Reply by nathantg** [reply]
Thanks, when I use the archived yamcs-data directory and run YAMCS, I notice that the size of the directory goes from about 1.5 GB to about 770MB. And every time I do it, the exact size of the directory gets reduced to a different size (1.5 GB to somewhere close to 770MB). Is this expected? Or could some data be lost?

Thanks for the helpful information.

**Reply by xpromache** [reply]
The size changes when you start Yamcs?
Or after rebuilding the parameter archive?

**Reply by nathantg** [reply]
When I start Yamcs. I was just curious to see why that happens when yamcs gets started. 

Could it be because the archived database includes databases for 3 instances (3 templates), but I am only using 1 instance when I start yamcs?

**Reply by xpromache** [reply]
There is a LOG file in the .rdb directory, maybe you can find something there to explain the drastic reduction in size. 
Yamcs would not remove any data just like that, no matter if you go from 3 instances to one instance.

How do you store the data of the 3 instances to the same RocksDB database? This can be done by specifying a tablespace option in the yamcs.instance.yaml but I see that option is not even properly documented.

**Reply by nathantg** [reply]
Thanks for the advice! I found that the size of the yamcs-data directory is being reduced when I start YAMCS because the LOG files and perhaps some other files are being deleted and overwritten. So not the actual database data size getting smaller. 

Is there a way to find the size of the database (actual archived data) in bytes/MB/GB?
